Logging + time zone issue

I log a QSO and the times that are inserted into the UI are local time. It should be UTC.

Workaround: I can correct the times myself before actually writing to the log. If I don't do that, the time zone entered into `js8call_log.adi` is also local time.

This is a Linux box, the box's clock actually runs UTC, but the environment variables are set appropriately for Germany, which is currently on UTC+2.

The JS8Call version is self-compiled from 902f9f8165, which is a bit later than v2.3.1.

Thanks for providing fine software!

@rruchte Rob, I tried to build from your branch logging-utc-date-fixes, and it failed with the same errors on both Linux Mint 22.1 and Win10 (JTSDK64-Tools), with the following errors:

```
C:\JTSDK64-Tools\tmp\wsjtx\logqso.cpp: In member function 'void LogQSO::on_start_now_button_pressed()':
C:\JTSDK64-Tools\tmp\wsjtx\logqso.cpp:71:24: error: 'class QDateTimeEdit' has no member named 'setTimeZone'; did you mean 'setTimeRange'?
   71 |   ui->start_date_time->[ 73%] setTimeZoneBuilding CXX object CMakeFiles/js8call.dir/varicode.cpp.obj(QTimeZone::utc());
      |
^~~~~~~~~~~
      |                        setTimeRange
C:\JTSDK64-Tools\tmp\wsjtx\logqso.cpp: In member function 'void LogQSO::on_end_now_button_pressed()':
In file included from C:\JTSDK64-Tools\tmp\wsjtx\logqso.cpp:76:24:C:\JTSDK64-Tools\tmp\wsjtx\varicode.cpp:27 :
error: C:/JTSDK64-Tools/tmp/wsjtx/vendor/CRCpp/CRC.h:895:' class QDateTimeEditwarning: [ 74%] ignoring '' has no member named '                                                      #pragma warning
setTimeZone' [                 -Wunknown-pragmassetTimeRange]
  895 | #   pragma warning (push)
      |
'?
   76 |   ui->start_date_time->                                                              (QTimeZone::utc());
      |
      |                                                    ' [
-Wunknown-pragmas
  896 | #   pragma warning (disable : 4333)
      |
C:\JTSDK64-Tools\tmp\wsjtx\logqso.cpp:C:/JTSDK64-Tools/tmp/wsjtx/vendor/CRCpp/CRC.h:900: In member function ' void LogQSO::initLogQSO(const QString&, const QString&, QString, const QString&, const QString&, const QDateTime&, const QDateTime&, Radio::Frequency, const QString&, const QString&, const QString&, const QString&)warning: ':
          C:\JTSDK64-Tools\tmp\wsjtx\logqso.cpp:227:24:#pragma warning     error: -Wunknown-pragmas'
  900 | #   pragma warning (pop)
      |
class QDateTimeEdit' has no member named 'setTimeZone'; did you mean 'setTimeRange'?
  227 |   ui->start_date_time->setTimeZone(QTimeZone::utc());
      |                        ^~~~~~~~~~~
      |                        setTimeRange
C:\JTSDK64-Tools\tmp\wsjtx\logqso.cpp:229:22: error: 'class QDateTimeEdit' has no member named 'setTimeZone'; did you mean 'setTimeRange[ 75%] Building CXX object CMakeFiles/js8call.dir/jsc_list.cpp.obj'?
  229 |   ui->end_date_time->
setTimeZone(QTimeZone::utc());
      |                      ^~~~~~~~~~~
      |                      setTimeRange
mingw32-make.exe[2]: *** [CMakeFiles\js8call.dir\build.make:342: CMakeFiles/js8call.dir/logqso.cpp.obj] Error 1
mingw32-make.exe[2]: *** Waiting for unfinished jobs....
```

73,
-Joe-
K0OG

--- a/logqso.ui
+++ b/logqso.ui
@@ -164,6 +164,9 @@
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
+             <property name="TimeSpec">
+              <enum>Qt::UTC</enum>
+             </property>
              <property name="displayFormat">
               <string>yyyy-MM-dd HH:mm:ss</string>
              </property>
@@ -218,6 +221,9 @@
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
+             </property>
+             <property name="TimeSpec">
+              <enum>Qt::UTC</enum>
              </property>
              <property name="displayFormat">
               <string>yyyy-MM-dd HH:mm:ss</string>

